src/serde_arrow_array_utils.erl

```diff
@@ -0,0 +1,60 @@
+%% @private
+%% Utilitary functions for Array Implementations.
+-module(serde_arrow_array_utils).
+-export([validity/2, buffer/2]).
+
+validity(Value, Len) ->
+    ValidityList = lists:map(
+        fun(X) ->
+            case X of
+                nil ->
+                    0;
+                _ ->
+                    1
+            end
+        end,
+        Value
+    ),
+    NullCount = Len - lists:sum(ValidityList),
+    case NullCount of
+        0 ->
+            {0, nil};
+        _ ->
+            {NullCount, bitmap(ValidityList, Len)}
+    end.
+
+bitmap(ValidityList, Len) ->
+    %% We need to split the list every 8th element and then pad the very last
+    %% element such that every resulting list has a length of 8 elements.
+    %%
+    %% If we try to split a list with only 8 elements, we will get a copy of the
+    %% list and an empty list in the resulting tuple. In order to get over this
+    %% empty list, we pad one bit extra and drop the final list.
+
+    PadLen = 9 - Len rem 8,
+    Padding = lists:duplicate(PadLen, 0),
+    SplitValidityList = lists:droplast(
+        tuple_to_list(lists:split(8, lists:append(ValidityList, Padding)))
```

Comment:
If I'm reading the comment right you want to split the list into segments of 8 elements? This looks to be splitting the list into 1 element of at most 8 elements and another of the rest.

Reply:
So the initial list may have any positive number (i.e >  0) of elements. We need to pad the list so that the number of elements is a multiple of 8. We then need to split the list into segments of 8 elements, where each element is made into a byte.

We are padding an extra bit to remove the empty list that comes as `List2` as a result of splitting a list of just 8 elements. We then can remove the extra bit, which is guaranteed to be extra regardless whether we are splitting a 7 element list or a 1025 element list.

The documentation as to *why* we are doing this can be found [here](https://arrow.apache.org/docs/format/Columnar.html#validity-bitmaps).

Reply:
I do want N / 8 lists. This does give me N / 8 lists. When I split an list of 24 elements, it give a three tuple.

Perhaps I should write a test case for this.

Reply:
Hm, then I'm missing something because `lists:split/2` returns a 2-tuple, always.

Reply:
You're right! When I tested this with a 24 element list I received incorrect output.
I'll get this fixed right study

Reply:
I believe a recursive version would be clearer and more efficient. Something like this:

```elixir
-define(validity(X), if X =:= nil or X =:= undefined -> 1; true -> 0 end).

something([X1, X2, X3, X4, X5, X6, X7, X8 | Rest], Acc, Count) ->
  B1 = ?validity(X1),
  B2 = ?validity(X2),
  B3 = ?validity(X3),
  B4 = ?validity(X4),
  B5 = ?validity(X5),
  B6 = ?validity(X6),
  B7 = ?validity(X7),
  B8 = ?validity(X8),

  something(
    Rest,
    <<Acc/binary, B1:1, B2:1, B3:1, B4:1, B5:1, B6:1, B7:1, B8:1>>,
    Count + B1 + B2 + B3 + B4 + B5 + B6 + B7 + B8
  ).
something([], Acc, Count) ->
  {Acc, Count};
something(LeftOver, Acc, Count) ->
  something(pad(LeftOver), Acc, Count).
```

It is a single pass, except for the padding. No closures, no high-order functions. Please give it a try.

Reply:
It is no problem, we are not expecting anyone to come up with the best solution upfront, the important is to listen, learn, and collaborate. :)